modules/freetype/src/freetype.cpp

```diff
@@ -723,10 +723,28 @@ int FreeType2Impl::coFn( const FT_Vector *cnt,
 
     PathUserData *p = (PathUserData *)user;
 
+    // Calicurate upper limit of CtoL.
+    // It is same as total length of polylines{ mOldP, cnt, to }.
+
+    const double dx1 = p->mOldP.x  -  cnt->x;
+    const double dy1 = p->mOldP.y  -  cnt->y;
+    const double dx2 = cnt->x      -  to->x;
+    const double dy2 = cnt->y      -  to->y;
+
+    const int maxL = static_cast<int>(
+        std::pow( dx1 * dx1 + dy1 * dy1, 0.5 ) +
+        std::pow( dx2 * dx2 + dy2 * dy2, 0.5 )
+    );
+
+    const int iCtoL = std::min( maxL , p->mCtoL );
+
+    // Avoid to push_back() same as previous points.
+    Point prevPt(INT_MIN, INT_MIN);
+
     // Bezier to Line
-    for(int i = 0;i <= p->mCtoL; i++){
+    for(int i = 0;i <= iCtoL; i++){
         // Split Bezier to lines ( in FreeType coordinates ).
-        double u = (double)i * 1.0 / (p->mCtoL) ;
+        double u = (double)i * 1.0 / (iCtoL) ;
```

Comment:
*1.0 is redundant.

Reply:
Yes, I fixed it. And I replaced from ` (double)i` that is C-style-cast,  to `static_cast<double>(i)`.